crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -56,84 +139,258 @@ impl TryFrom<&JpmorganRouterData<&PaymentsAuthorizeRouterData>> for JpmorganPaym
     ) -> Result<Self, Self::Error> {
         match item.router_data.request.payment_method_data.clone() {
             PaymentMethodData::Card(req_card) => {
+                if item.router_data.is_three_ds() {
+                    return Err(errors::ConnectorError::NotSupported {
+                        message: "Three_ds payments".to_string(),
+                        connector: "Jpmorgan",
+                    }
+                    .into());
+                }
+                let capture_method = if let Some(method) = item.router_data.request.capture_method {
+                    map_capture_method(method)
+                } else {
+                    String::from("AUTOMATIC")
+                };
+
+                let currency = item.router_data.request.currency.to_string();
+
+                let merchant_software = JpmorganMerchantSoftware {
```

Comment:
no need to create unnecessary variables can directly be assigned to respective field while returning.